service/src/main/java/eu/europa/ec/fisheries/uvms/movement/service/dao/MovementDao.java

```diff
@@ -225,6 +231,46 @@ public Long getMovementListSearchCount(String countSql, List<SearchValue> search
         return query.getSingleResult();
     }
 
+    public List<Long> findMovementAreaIdsByAreaRemoteIdAndNameList(List<String> remoteIds,List<String> areaNames)  {
+
+        TypedQuery<Long> typedQuery = em.createNamedQuery(Movementarea.FIND_AREA_IDS_BY_AREA_REMOTEID_AND_NAME, Long.class);
+        typedQuery.setParameter("remoteIds",remoteIds);
+        typedQuery.setParameter("areaNames", areaNames);
+
+        return typedQuery.getResultList();
+    }
+
+    public boolean checkMovementExistence(String connectId, Date startDate, Date endDate, List<Long> movementAreaIds)  {
+        // current named query Movement.FIND_BY_CONNECTID_FOR_AREAS_IN_A_PERIOD
```

Comment:
Take a note of the alternative implementations, but remove this commented out code!